src/main/scala/main/solutions/MazeSolver/Helpers.scala

```diff
@@ -0,0 +1,25 @@
+package main.solutions.MazeSolver
+
+import scala.collection.mutable.HashSet
+
+object Helpers {
+  private def filterInBoundNeighbors(maze: Array[Array[Int]], point: Point): Boolean = {
```

Comment:
i wouldn't implement like that, you are a bit abusing the exception. Instead you could have just check that the point are not outside the array length.